### ISSUE TITLE
Check more FC/F77 info in configure

### ISSUE THREAD

**Issue by RonRahaman** [libMesh/libmesh#1703]
My MOOSE app wraps an existing F77 physics app.  It uses as much info as possible from `liibmesh-config`, but still needs to run its own autoconf script to get some more compiler info.  Would you consider adding some of these compiler checks to the libmesh autoconf script?

The checks I have to do are:
* `AC_F77_WRAPPERS`: discovers name-mangling scheme for calling F77 functions in C
* `AC:_FC_PP_DEFINE`: discovers the flag for defining preprocessor symbols (e.g, `-D` or `-Df`)
* `AC_FC_PP_SRCEXT([f])`: discovers the flag for preprocessing `.f` files.

I think the first two are commonly-used by well-coded Fortran projects and should definitely be considered.  

Even though my project needs last macro (`AC_FC_PP_SRCEXT`), it shouldn't be necessary for projects coded with the correct source file extensions.  Almost universally, compilers will preprocess `.F` files but not `.f` files.  Same with `.F90` and `.f90` files.  The original developers of my physics app weren't aware of this distinction.

**Comment by jwpeterson** [comment]
Yes, this certainly seems reasonable. For some reason I was not aware of `AC_F77_WRAPPERS` but it seems quite useful.

**Comment by RonRahaman** [comment]
Great!  Let me know if I can help or propose a Pull Request or whatnot.

**Comment by roystgnr** [comment]
I'd love a PR, if you would.  I'd also be happy to see these added to our configure tests, but since I don't have anything that would use the result I'd be flying blind if I added them myself.

**Comment by jwpeterson** [comment]
> I'd be flying blind if I added them myself.

Agreed. You might put the new function calls directly in `configure.ac`, right after the `LIBMESH_SET_CXX_FLAGS` call. After changing this file, you'll need to run `./bootstrap` using the libmesh-specific versions of the autotools. There is some [documentation](https://github.com/libMesh/libmesh/wiki/Updating-configure-tests%2C-Makefiles%2C-etc.) for this on our wiki.
